### PR TITLE
Script for doing release locally

### PR DESCRIPTION
**Pull request by beeender**
There are still more work needed to do release on Jenkins. Before that a
script to help release locally is useful.
This script currently only handle the patch release.

### CODE REVIEW

File: tools/release.sh [hunk 1]
```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env bash
+
```
Comment (cmelchior) [inline tools/release.sh]: Perhaps add a comment here that this only works for patch releases

File: tools/release.sh [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    echo "Building..."
+
+    ./gradlew assemble
+
+    echo "Verifying examples..."
+
+    check_adb_device
+
+    # Verify examples
+    cd examples
```
Comment (kneth) [inline tools/release.sh]: Maybe `(cd examples && ./gradlew uninstallAll && ./gradlew monkeyDebug)` and no `cd ..`

File: tools/release.sh [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+    echo "Done."
+    echo "1. Log into BinTray(https://bintray.com) with the Realm account;"
+    echo "2. Goto https://bintray.com/realm/maven and check if there are 16 artifacts to publish."
+    echo "3. Press 'Publish'."
+    while true
+    do
+        read -r -p "Have you published 16 artifacts on Bintray? Type 'Yes' to continue... " input
+
+        case $input in
```
Comment (kneth) [inline tools/release.sh]: `"$input"` is safer

File: tools/release.sh [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    echo "Publishing distribution package..."
+
+    # Create distribution package
+    ./gradlew distributionPackage
+    cd build/outputs/distribution
+    unzip realm-java-${VERSION}.zip
+
+    # Test
+    check_adb_device
+    cd examples/
```
Comment (kneth) [inline tools/release.sh]: Consider to use `pushd` and `popd`.

**Comment by emanuelez** [comment]
`shellcheck` found 17 warnings. Nothing too serious, but nonetheless it shows some best practices that won't hurt.

File: tools/release.sh [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        echo "Cannot find executable file 's3cmd'."
+        exit -1
+    fi
+    if [[ ! -e $HOME/.s3cfg ]] ; then
+        echo "'$HOME/.s3cfg' cannot be found."
+        exit -1
+    fi
+
+    # Check BinTray credentials
+    if ! grep "bintrayUser=realm" $HOME/.gradle/gradle.properties > /dev/null ; then
```
Comment (beeender) [inline tools/release.sh]: I prefer without `-c` :P

**Pull request merged**